slides/A2-advanced-intro.md

```diff
@@ -423,6 +423,23 @@ enum IpAddressType {
 * Each variant is an alternative value of the enum, you pick a single value to
   create an instance
 
+---
+
+# Enumerations
+One of the more powerful kinds of types in Rust are enumerations
+
+```rust
+enum IpAddressType {
+  Ipv4, // = 0 (default discriminant)
+  Ipv6, // = 1 (default discriminant)
+}
+```
+
+* An enumeration (listing) of different *variants*
+* Each variant is an alternative value of the enum, you pick a single value to
+  create an instance
+* Each variant has a discriminant (hidden by default), a numeric value (`isize` by default) used to determine the variant that the enumeration holds
```

Comment:
It'd be good to note that the discriminant may be smaller due to alignment or optimization. I think at this point in the course it's good to just state that there's a numerical discriminant, and that it may vary in size or type. 

If instead you want to make the discriminant type explicit, then please add some content on how one would manipulate the type of the discriminant (`#[repr(<ty>)]) and what considerations should be made when doing so. Also, it's best to note that one cannot rely on the actual type of the descriminants unless it's explicity defined

Reply:
I added some extra information about this. Would this work now or should I add an extra slide?

Reply:
Looks good, thanks!